pkg/cloud/ovirt/machine/actuator.go

```diff
@@ -281,6 +283,11 @@ func (actuator *OvirtActuator) reconcileNetwork(machine *machinev1.Machine, inst
 	// expect IP addresses only on those statues.
 	case ovirtsdk.VMSTATUS_UP, ovirtsdk.VMSTATUS_MIGRATING, ovirtsdk.VMSTATUS_POWERING_UP:
 		break
+
+	// while vm is rebooting retry reconciling
+	case ovirtsdk.VMSTATUS_REBOOT_IN_PROGRESS:
+		return fmt.Errorf("Skipping network reconciliation while VM %s is rebooting", instance.MustName())
```

Comment:
I think it's worth logging it here for other states too (in addition to L249). We should make it clear why it is being skipped.
More importantly, it needs to be really explained better when we do want to reconcile and when we don't.

I would prefer if it's explicit and we proceed only in Up/Migrating. Those are the only two states when VM is running and we can be reasonably sure that everything is up and running. PoweringUp is questionable....but if this gets triggered only as a result of kubelet event it means it's already running, then it's fine. But if it gets triggered earlier by other means then it can easily happen when VM is not booted yet.
Either way, please clarify in a comment

Also please clarify how it should behave in case of states where we don't expect it to come up - PoweringDown, Down and such - what should happen? Keep retrying reconciliation or give up? What's the effect of lines 291-293 when status is e.g. "Down" (for whatever reason). It will never recover from OCP POV. But for some states (Paused) you can't really tell if it's temporary or final (paused for storage extension is temporary, paused for storage error is likely final)...should we keep trying?